Remove outdated assertion on model support

liam-sbhoo opened the pull request:
### Change Description

I think this assertion is outdated.

**Addition / Fix:**
- `TabPFNRegressor.list_available_models()`
- add model support
- fix unnecessary welcoming message everytime we call fit.

**I have tested it (checked from Vertex log, make sure the model specified in client is being used).**

*Try to be precise. You can additionally add comments to your PR, this might help the reviewer a lot.*


**If you used new dependencies: Did you add them to `requirements.txt`?**

NO

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**

## Breaking changes

If you made any breaking changes, please update the version number.
Breaking changes are totally fine, we just need to make sure to keep the users informed and the server in sync.

**Does this PR break the API? If so, what is the corresponding server commit?**

**Does this PR break the user interface? If so, why?**

---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

SamuelGabriel reviewed (approved):
LGTM

liam-sbhoo commented on tabpfn_client/tests/unit/test_tabpfn_classifier.py:
```diff
@@ -58,10 +58,6 @@ def test_init_remote_classifier(
             mock_server.endpoints.retrieve_greeting_messages.path
         ).respond(200, json={"messages": []})
 
-        mock_server.router.get(mock_server.endpoints.protected_root.path).respond(
```
Ideally, we don't need this here.
This line used to pass because we are calling `init()` multiple times, and the subsequent `init()` will call this API.

liam-sbhoo replied:
@SamuelGabriel

SamuelGabriel replied:
what do you mean? we don't need the "protected" call? I think so too as almost any action will do a protected call. removing it will save latency. do you mean that?

liam-sbhoo replied:
This API is a dummy endpoint that requires authentication. One use case is to check the user can login successfully or not. All the other API doesn't go through.

liam-sbhoo commented on tabpfn_client/tests/unit/test_tabpfn_regressor.py:
```diff
@@ -56,10 +56,6 @@ def test_init_remote_regressor(
             mock_server.endpoints.retrieve_greeting_messages.path
         ).respond(200, json={"messages": []})
 
-        mock_server.router.get(mock_server.endpoints.protected_root.path).respond(
```
Same as https://github.com/automl/tabpfn-client/pull/36/files#r1760022912

SamuelGabriel commented on tabpfn_client/estimator.py:
```diff
@@ -232,9 +232,17 @@ def predict_proba(self, X):
     def __init__(
         self,
-        model: str = "latest_tabpfn_hosted",
+        model_path: str = "default",
```
can you please stick to the "model" name

liam-sbhoo commented:
I renamed `model_path` to `model` (although some pre-processing required, TabPFN takes both `model` and `model_path` and they meant different things).
Added some test case too.
